pkg/operator/ceph/cluster/watcher.go

```diff
@@ -130,6 +144,319 @@ func (c *clientCluster) onK8sNode(ctx context.Context, object runtime.Object) bo
+		logger.Infof("rbd PV NAME %v", pv.Spec.CSI.VolumeAttributes)
+		args := []string{"status", fmt.Sprintf("%s/%s", pv.Spec.CSI.VolumeAttributes["pool"], pv.Spec.CSI.VolumeAttributes["imageName"])}
+		cmd := cephclient.NewRBDCommand(c.context, clusterInfo, args)
+		cmd.JsonOutput = true
+
+		buf, err := cmd.Run()
+		if err != nil {
+			return fmt.Errorf("failed to list watchers for pool/imageName %s/%s. %v", pv.Spec.CSI.VolumeAttributes["pool"], pv.Spec.CSI.VolumeAttributes["imageName"], err)
+		}
+
+		logger.Infof("rbd status to get the client ips %v", string(buf))
+		ips, err := rbdStatusUnMarshal(buf)
+		if err != nil {
+			return fmt.Errorf("failed to unmarshal rbd status output. %v", err)
+		}
+		err = c.createNetworkFence(ctx, pv, node, cluster, ips)
+		if err != nil {
+			return fmt.Errorf("failed to create network fence for node %q. %v", node.Name, err)
+		}
+	}
+	return nil
+}
+
+func rbdStatusUnMarshal(output []byte) ([]string, error) {
+	type rbdStatus struct {
+		Watchers []struct {
+			Address string `json:"address"`
+		} `json:"watchers"`
+	}
+
+	var rbdStatusObj rbdStatus
+	err := json.Unmarshal([]byte(output), &rbdStatusObj)
+	if err != nil {
+		return []string{}, fmt.Errorf("failed to unmarshal rbd status output. %v", err)
+	}
+
+	watcherIPlist := []string{}
+	for _, watcher := range rbdStatusObj.Watchers {
+		// split with separation '/' to remove nounce and concatenating `/32` to define a network with only one IP address
+		address := strings.Split(watcher.Address, "/")[0] + "/32"
+		watcherIPlist = append(watcherIPlist, address)
+	}
+	return watcherIPlist, nil
+}
+
+func (c *clientCluster) fennceCephFSVolumes(
+	ctx context.Context, node corev1.Node, cluster *cephv1.CephCluster,
+	clusterInfo *cephclient.ClusterInfo, listCephFSPV []corev1.PersistentVolume) error {
+
+	for _, pv := range listCephFSPV {
+		logger.Infof("cephfs PV NAME %v", pv.Spec.CSI.VolumeAttributes)
+
+		status, err := cephclient.StatusWithUser(c.context, clusterInfo)
+		if err != nil {
+			return fmt.Errorf("failed to get ceph status for check active mds. %v", err)
+		}
+
+		var activeMDS string
+		for _, fsRank := range status.Fsmap.ByRank {
+			if fsRank.Status == "up:active" {
+				activeMDS = fsRank.Name
+			}
+		}
+
+		args := []string{"tell", fmt.Sprintf("mds.%s", activeMDS), "client", "ls", "--format", "json"}
+		cmd := cephclient.NewCephCommand(c.context, clusterInfo, args)
+		cmd.JsonOutput = true
+
+		buf, err := cmd.Run()
+		if err != nil {
+			return fmt.Errorf("failed to list watchers for cephfs pool/subvoumeName %s/%s. %v", pv.Spec.CSI.VolumeAttributes["pool"], pv.Spec.CSI.VolumeAttributes["subvolumeName"], err)
+		}
+		ips, err := cephFSMDSClientMarshal(buf, pv)
+		if err != nil || ips == nil {
+			return fmt.Errorf("failed to unmarshal cephfs mds  output. %v", err)
+		}
+
+		err = c.createNetworkFence(ctx, pv, node, cluster, ips)
+		if err != nil {
+			return fmt.Errorf("failed to create network fence for node %q. %v", node.Name, err)
+		}
```

Comment:
Same here as well. IF you have both cephfs and RBD PVC you just need to fine either of them not both

Reply:
just to make clear myself, if I have multiple rbd/cephfs volume in a particular node, we just need to block only one IP it can be for cephfs or rbd?

Reply:
@Madhu-1 ^^

Reply:
Yes that correct

Reply:
ohh then it's made easier for me. Thanks